File: content/snippets/typescript/exercises/advanced/activity_selection_problem.ts
const acts: [number,number][] = [[1,4],[3,5],[0,6],[5,7],[8,9],[5,9]]
acts.sort((a,b)=>a[1]-b[1])
const res: [number,number][] = []
let last=-1
for(const [s,e] of acts) if(s>last){ res.push([s,e]); last=e }
console.log('Selected', res)
export {}

File: content/snippets/cpp/exercises/advanced/activity_selection_problem.cpp
#include <bits/stdc++.h>
using namespace std;int main(){ vector<pair<int,int>> acts={{1,4},{3,5},{0,6},{5,7},{8,9},{5,9}}; sort(acts.begin(), acts.end(), [](auto &a, auto &b){ return a.second < b.second; }); int last= -1; cout<<"Selected:"; for(auto &p:acts) if(p.first>last){ cout<<" ("<<p.first<<","<<p.second<<")"; last=p.second; } cout<<"\n"; }
